config/deploy/docker/lib/cronjob.rb

```diff
@@ -0,0 +1,152 @@
+# An abstraction around a kubernetes cronjob
+# https://github.com/k8s-ruby/k8s-ruby
+class Cronjob
+  attr_accessor :args
+  attr_accessor :capacity_type
+  attr_accessor :description
+  attr_accessor :jobname
+  attr_accessor :schedule_expression
+  attr_accessor :k8s_image
+  attr_accessor :k8s_service_account_name
+
+  MAX_NAME_LENGTH = 52
+
+  def initialize(description:, schedule_expression:, command:, capacity_type: 'spot')
```

Comment:
I haven't found it yet, so just noting, `capacity_provider` should probably not be spot if the task is not interruptible.

Reply:
I'm just making the default the cheaper option. The logic in `config/schedule.rb` is explicit about all of the entries I think. interruptible jobs are put on spot capacity. non-interruptible on on-demand.